api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseRelatedPersonDao.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.dao.impl;
+
+import static org.hibernate.criterion.Restrictions.and;
+import static org.hibernate.criterion.Restrictions.eq;
+import static org.hibernate.criterion.Restrictions.isNull;
+import static org.hibernate.criterion.Restrictions.or;
+import static org.hibernate.criterion.Restrictions.sqlRestriction;
+import static org.hl7.fhir.r4.model.Patient.SP_FAMILY;
+import static org.hl7.fhir.r4.model.Patient.SP_GIVEN;
+import static org.hl7.fhir.r4.model.Person.SP_ADDRESS_CITY;
+import static org.hl7.fhir.r4.model.Person.SP_ADDRESS_COUNTRY;
+import static org.hl7.fhir.r4.model.Person.SP_ADDRESS_POSTALCODE;
+import static org.hl7.fhir.r4.model.Person.SP_ADDRESS_STATE;
+import static org.hl7.fhir.r4.model.Person.SP_BIRTHDATE;
+import static org.hl7.fhir.r4.model.Person.SP_NAME;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+import org.hibernate.Criteria;
+import org.hibernate.criterion.DetachedCriteria;
+import org.hibernate.criterion.Order;
+import org.hibernate.criterion.Projections;
+import org.hibernate.criterion.Subqueries;
+import org.hibernate.sql.JoinType;
+import org.openmrs.Auditable;
+import org.openmrs.OpenmrsObject;
+import org.openmrs.PersonName;
+
+public abstract class BaseRelatedPersonDao<T extends OpenmrsObject & Auditable> extends BaseFhirDao<T> {
```

Comment:
Does this need to be in a separate class? The reason `BasePersonDao` is separated out is because there are a number of resources that all use `Person` and need to have similar logic, but the logic here seems pretty specific to `RelatedPerson`. Or is there something I'm missing?

Reply:
Hi @ibacher  I think yes since the relationship is not directly person. So inside relation yes we have personA which is the related person in fhir. The queries are a bit different as you can see.

Reply:
Hi @ibacher  I think atleast we have to get the use of BasePersonDao class. Since we need to have all the personrelated sort stuff in the relatedperson as well.

but anyway if we are not maintaining a sperate class I think its better to alter the 
handleSort in the base dao to include a person alisa parameter as we did for the name search/ 
wdyt?

Reply:
@ibacher  waiting for youre response on this

Reply:
I just meant "do we need a `BaseRelatedPersonDao` class or can we move all of this into `RelatedPersonDao`?" We aren't really reusing `BasePersonDao` here at all.

Reply:
oh yes sorry about this @ibacher